services/app-web/src/pages/SubmissionRevisionSummary/SubmissionRevisionSummaryV2.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { useEffect } from 'react'
+import { GridContainer } from '@trussworks/react-uswds'
+import { useParams } from 'react-router-dom'
+import { Loading } from '../../components/Loading'
+import { useAuth } from '../../contexts/AuthContext'
+import { ContractDetailsSummarySectionV2 } from '../StateSubmission/ReviewSubmit/V2/ReviewSubmit/ContractDetailsSummarySectionV2'
+import { ContactsSummarySection } from '../StateSubmission/ReviewSubmit/V2/ReviewSubmit/ContactsSummarySectionV2'
+import { RateDetailsSummarySectionV2 } from '../StateSubmission/ReviewSubmit/V2/ReviewSubmit/RateDetailsSummarySectionV2'
+import { SubmissionTypeSummarySectionV2 } from '../StateSubmission/ReviewSubmit/V2/ReviewSubmit/SubmissionTypeSummarySectionV2'
+import { usePage } from '../../contexts/PageContext'
+import { GenericErrorPage } from '../Errors/GenericErrorPage'
+import { Error404 } from '../Errors/Error404Page'
+import { dayjs } from '../../common-code/dateHelpers'
+import styles from './SubmissionRevisionSummary.module.scss'
+import { PreviousSubmissionBanner } from '../../components'
+import {
+    useFetchContractQuery,
+} from '../../gen/gqlClient'
+import { ErrorForbiddenPage } from '../Errors/ErrorForbiddenPage'
+
+type RouteParams = {
+    id: string
+    revisionVersion: string
+}
+
+export const SubmissionRevisionSummaryV2 = (): React.ReactElement => {
+    // Page level state
+    const { id } = useParams<keyof RouteParams>()
+    if (!id) {
+        throw new Error(
+            'PROGRAMMING ERROR: id param not set in state submission form.'
+        )
+    }
+    const { updateHeading } = usePage()
+    const { loggedInUser } = useAuth()
+
+    const isStateUser = loggedInUser?.role === 'STATE_USER'
+    const isCMSUser = loggedInUser?.role === 'CMS_USER'
+
+    const {
+        data: fetchContractData,
+        loading: fetchContractLoading,
+        error: fetchContractError,
+    } = useFetchContractQuery({
+        variables: {
+            input: {
+                contractID: id ?? 'unknown-contract',
```

Comment:
I think we are missing some key logic here -- - assuming we are copying previous behavior the `id` on the route params here  - this URL param not the `id` of the contract it is a number that represents the number of this revisions  (e.g. a number like 1 for first ,2 for second ,3 for third , etc etc). 

 If the intent here is not clear - let's talk.  Its a weird thing we only do for urls for previous submissions and not where else - you would have to look at ChangeHistory and SubmissionRevisionSummary side by side to see how these URLs work.

Reply:
We can meet to talk about this, because I'm seeing that the route contains the id `/submissions/:id/revisions/:revisionVersion` and that the previous SubmissionRevisionSummary also pulled the id out to fetch the hpp

I think you're right that I was missing pulling out the `revisionNumber`. It's needed to find the index of the submission from packageSubmissions

Reply:
Makes sense. Reading it this morning. my comment wasn't quite right 😮‍💨 😆 either. Yes the revision version matters also is the point.